crates/suiop-cli/src/cli/ci/image.rs

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::cli::lib::{get_api_server, get_oauth_token};
+use anyhow::Result;
+
+use chrono::{DateTime, Local, Utc};
+use clap::{Parser, ValueEnum};
+use colored::Colorize;
+use serde::{self, Serialize};
+use std::{fmt::Display, str::FromStr};
+use tabled::{settings::Style, Table, Tabled};
+use tracing::debug;
+
+#[derive(Tabled)]
+struct BuildInfo {
+    name: String,
+    status: String,
+    #[tabled(rename = "Start Time (Local Time)")]
+    start_time: String,
+    #[tabled(rename = "End Time")]
+    end_time: String,
+}
+
+#[derive(Parser, Debug)]
+pub struct ImageArgs {
+    #[command(subcommand)]
+    action: ImageAction,
+}
+
+#[derive(ValueEnum, Clone, Debug)]
+#[clap(rename_all = "lowercase")]
+pub enum RefType {
+    Branch,
+    Tag,
+    Commit,
+}
+
+impl Serialize for RefType {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(match self {
+            RefType::Branch => "branch",
+            RefType::Tag => "tag",
+            RefType::Commit => "commit",
+        })
+    }
+}
+
+impl Display for RefType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            RefType::Branch => write!(f, "branch"),
+            RefType::Tag => write!(f, "tag"),
+            RefType::Commit => write!(f, "commit"),
+        }
+    }
+}
+
+#[derive(clap::Subcommand, Debug)]
+pub enum ImageAction {
+    #[command(name = "build")]
+    Build {
+        #[arg(short, long)]
+        repo_name: String,
+        #[arg(short, long)]
+        dockerfile: String,
+        #[arg(long)]
```

Comment:
todo: incorporate automatic version incrementing at some point
```suggestion
        /// The tag for this build
        #[arg(long)]
```

Reply:
on the server side, after cloning the app repo, it will automatically add the checkout sha as a tag, so this is not a required field. It's more or less like a semantic image tag that people can use optionally.

Reply:
I'll modify the wording and add a comment line, thanks!